### PR TITLE
OZ-554: E2E tests for handling voided entities when streaming with Analytics

### PR DESCRIPTION
**Pull request by kdaud**
Ticket [OZ-554](https://mekomsolutions.atlassian.net/browse/OZ-554)

[OZ-554]: https://mekomsolutions.atlassian.net/browse/OZ-554?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

This PR adds the following E2E tests handling voidable entities when streaming with Analytics.

- Voiding an OpenMRS patient updates the patient in Superset patients table.
- Voiding an OpenMRS condition updates the condition in Superset conditions table.
- Voiding an OpenMRS encounter updates the encounter in Superset encounters table.
- Cancelling an OpenMRS appointment updates the appointment in Superset appointments table.

### CODE REVIEW

File: e2e/tests/openmrs-superset-flows.spec.ts [hunk 1]
```diff
@@ -419,9 +441,179 @@ test('Voiding an OpenMRS obs updates the observation in Superset observations ta
...
+  await superset.clearSQLEditor();
+  let personVoidedQuery = `SELECT person_voided FROM patients where identifiers like 'OpenMRS ID: ${patientIdentifier}';`;
+  await page.getByRole('textbox').first().fill(personVoidedQuery);
+  await superset.runSQLQuery();
+  let patientVoidedState = await page.locator('div.virtual-table-cell');
+
+  await expect(patientVoidedState).toContainText('false');
+
+  // replay
+  await openmrs.deletePatient();
```
Comment (kdaud) [inline e2e/tests/openmrs-superset-flows.spec.ts:464]: This particular test case invokes `openmrs.deletePatient()` during the replay thus voiding the patient. This implies that the method can not be called again in the cleanup function (afterEach()) with this particular test so the suggested approach is to let each test case invoke the function that voids  the patient.

Comment (Ruhanga) [reply e2e/tests/openmrs-superset-flows.spec.ts:464]: If deletions are actually voids in OpenMRS, I'd rather change `openmrs.deletePatient()` to `openmrs.voidPatient()`.

**Review by Ruhanga — COMMENTED** [review]
Generally LGTM.

File: e2e/utils/functions/openmrs.ts [hunk 2]
```diff
@@ -145,15 +145,14 @@ export class OpenMRS {
     await this.page.getByRole('button', { name: 'Close' }).click();
   }
 
-  async deletePatient() {
+  async voidPatient() {
     await this.page.goto(`${O3_URL}/openmrs/admin/patients/index.htm`);
     await this.page.getByPlaceholder(' ').type(`${patientName.firstName + ' ' + patientName.givenName}`);
     await this.page.locator('#openmrsSearchTable tbody tr.odd td:nth-child(1)').click();
     await this.page.locator('input[name="voidReason"]').fill('Delete patient created by smoke tests');
```
Comment (Ruhanga) [inline e2e/utils/functions/openmrs.ts:152]:
```suggestion
    await this.page.locator('input[name="voidReason"]').fill('Void patient created by smoke tests');
```

**Review by Ruhanga — COMMENTED** [review]
This has to change too?

**Review by Ruhanga — APPROVED** [review]
LGTM!

**Pull request merged**